src/app/content/components/Page.spec.tsx

```diff
@@ -804,6 +820,58 @@ describe('Page', () => {
     expect(scrollTo).toHaveBeenCalledWith(highlightElement);
   });
 
+  it('doesn\'t render error modal for the same result twice', async() => {
+    const {root} = renderDomWithReferences();
+
+    // page lifecycle hooks
+    await Promise.resolve();
+
+    const highlightResults = jest.spyOn(searchUtils, 'highlightResults');
+    const hit = makeSearchResultHit({book, page});
+    const searchResultToSelect = {result: hit, highlight: 0};
+
+    highlightResults.mockReturnValue([]);
+
+    store.dispatch(requestSearch('asdf'));
+    store.dispatch(receiveSearchResults(makeSearchResults([hit])));
+    store.dispatch(selectSearchResult(searchResultToSelect));
+
+    // page lifecycle hooks
+    await Promise.resolve();
+    // after images are loaded
+    await Promise.resolve();
+
+    const errorModalCloseButton = root.querySelector('[data-testid=banner-body] button');
+
+    if (!errorModalCloseButton) {
+      return expect(errorModalCloseButton).toBeTruthy();
+    }
+
+    renderer.act(() => {
+      ReactTestUtils.Simulate.click(errorModalCloseButton);
+    });
+
+    expect(root.querySelector('[data-testid=banner-body]')).toBeFalsy();
+
+    jest.spyOn(highlightUtils, 'highlightData').mockReturnValue(() => ({} as any));
+
+    // normally, search result selection handler would noop if the
+    // search result is the same. This makes it think that a new highlight was
+    // added and will force reselection
+
+    renderer.act(() => {
+      store.dispatch(createHighlight({} as any, {} as any));
+      store.dispatch(selectSearchResult(searchResultToSelect));
+    });
```

Comment:
I'm not sure why those two dispatches show a warning that they are not inside a `renderer.act`